decapodes.it

```diff
@@ -0,0 +1,137 @@
+@sum Parameter begin
+  LiteralParameter(type::String, value::Float64)
+  ConstantParameter(type::String, value::String)
+end
+
+struct Condition
+  type::String
+  value::String
+  domain_mesh::String
+end
+
+struct DatasetFile
+	uri::String
+	format::String
+  shape::Vector{Int64}
+end
+
+struct Dataset
+  type::String
+  name::String
+  description::String
+  file::DatasetFile  
+end
+
+@sum MeshValue begin
+  File(uri::String, checksum::String, format::String, shape::Vector{Int64})
+  Values(values::Vector{Int64})
+end
+
+struct DecapodeConfiguration
+  parameters::Object{Parameter}
+  initial_conditions::Object{Condition}
+  boundary_conditions::Object{Condition}
+  datasets::Object{Dataset}
+end
+
+struct ASKEMDecapodeConfiguration
+  header::AMRBase.ConfigurationHeader
+  configuration::DecapodeConfiguration
+end
+
+@sum Constant begin
+	FileConstant(type::String, value::AMRBase.File)
+  RawConstant(type::String, value::Float64)
+end
+
+@sum ConstraintValue begin
+	UnitValue(value::Float64, unit::String)
+	Datetime(timestamp::String)
+end
+
+struct Dimensionality
+	manifold::Int64
+	embedding_space::Int64
+end
+
+@sum Region begin
+	RegionEquation(equation::String)
```

Comment:
Algebraic varieties are determined by a system of equations so it should be a list.